src/organisms/gv-pagination/gv-pagination.ts

```diff
@@ -0,0 +1,324 @@
+ */
+@customElement('gv-pagination')
+export class GvPagination extends LitElement {
+  @property({ type: Object })
+  public data: Pagination | any;
+
+  @property({ type: Boolean, attribute: 'hide-empty' })
+  public hideEmpty: boolean = false;
+
+  @property({ type: Boolean })
+  public widget: boolean = false;
+
+  @property({ type: Boolean })
+  public disabled: boolean = false;
+
+  @property({ type: Boolean, attribute: 'has-search' })
+  public hasSearch: boolean = false;
+
+  @property({ type: Number, attribute: false })
+  private max: number = 10;
+
+  @property({ type: Boolean, attribute: 'has-select' })
+  public hasSelect: boolean = false;
+
+  @property({ type: Boolean, reflect: true })
+  public small: boolean = true;
+
+  @property({ type: Boolean, reflect: true })
+  public medium: boolean = false;
+
+  @property({ type: Boolean, reflect: true })
+  public large: boolean = false;
+
+  @property({ type: Number, attribute: false })
+  private center: number = this.max / 2 - 1;
+
+  @query('gv-input')
+  private input: any;
+
+  static get styles() {
+    return [
+      // language=css
+      css`
+        .pagination {
+          --font-size: var(--gv-pagination--fz, var(--gv-theme-font-size-s, 12px));
+          font-size: var(--fz);
+          --gv-button--fz: var(--fz);
+          display: flex;
+          align-items: center;
+        }
+
+        gv-button {
+          --gv-icon--s: var(--gv-pagination-icon--s, 18px);
+          min-width: 29px;
+        }
+
+        gv-input {
+          width: 50px;
+        }
+
+        .medium gv-input,
+        .large gv-input {
+          width: 60px;
+        }
+
+        gv-select-native {
+          width: 65px;
+        }
+
+        .medium gv-select-native,
+        .large gv-select-native {
+          width: 75px;
+        }
+      `,
+    ];
+  }
+
+  willUpdate(changedProperties: any) {
+    if (changedProperties.has('data') && this.data) {
+      this.max = Math.min(this.data.total_pages, 10);
+    }
+    if (changedProperties.has('max') && this.max != null) {
+      this.center = this.max / 2 - 1;
+    }
+
+    if (changedProperties.has('large') && this.large) {
+      this.small = false;
+      this.medium = false;
+    } else if (changedProperties.has('medium') && this.medium) {
+      this.small = false;
+      this.large = false;
+    } else if (changedProperties.has('small') && this.small) {
+      this.medium = false;
+      this.large = false;
+    }
+  }
+
+  private goToPage(page: number) {
+    this.data.current_page = page;
+    this.requestUpdate('data');
+    dispatchCustomEvent(this, 'paginate', { page: page, size: this.data.size } as PaginateDetail);
+  }
+
+  private onSubmit(e: CustomEvent<number>) {
+    const page = e.detail;
+    if (page != null && page <= this.data.total_pages && page > 0) {
+      this.goToPage(page);
+    }
+  }
+
+  private onClickToSearch() {
+    if (this.input) {
+      // @ts-ignore
```

Comment:
still necessary ? input is any so it should be good?

Reply:
✅